bundler/lib/bundler/settings.rb

```diff
@@ -316,7 +318,7 @@ def key_for(key)
     private
 
     def configs
-      {
+      @configs ||= {
```

Comment:
is there ever a time when we would want to recompute this ivar?

Reply:
alternatively, what if this became an `each_config` method, yielding the key, value pairs? then we dont need the hash at all, if all we ever do is iterate over it

Reply:
> is there ever a time when we would want to recompute this ivar?

I gave this some thought initally, but forgot to note it. I don't think so though.

My initial thought that was that it wouldn't need to reset if the contents of the ivars in it change (ie `@temporary` and all). On closer review, the contents of the ivars could change (ie `@temporary = {}`), but I don't see any writer methods or any code that assigns directly to them.

> alternatively, what if this became an each_config method, yielding the key, value pairs? then we dont need the hash at all, if all we ever do is iterate over it

Oh, that is an interesting thought! I tested it, and yeah, it saves allocating a hash and is faster 😮  Updated my benchmark with it https://gist.github.com/technicalpickles/91933d34dcd47bfc99d6ac0e53fcdb1e (as `correct manual_loop w/ each_config`)